src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1291,6 +1424,72 @@ def construct_payload(self):
         super().construct_payload()
         self.set_up_service_bindings()
         self.set_up_unbind_service_bindings()
+        self.set_up_source()
+
+    def set_up_source(self):
+        if self.get_argument_source():
```

Comment:
does your feature only work for managed environment? If yes, please add validation for --environment-type. 
 For more information, @Greedygre can help.

Reply:
This feature only work for managed environment. Because it use a lot of class from `up`, which only support managed environment. 
@snehapar9 
please need to add validation for `--environment-type`, if `environement-type` value is `connected`, or `managed_env` value is a connected environment reource id, should not use `--repo` and `--source`.

Reply:
Updated.